pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -638,19 +638,44 @@ func (m *kubeGenericRuntimeManager) pruneInitContainersBeforeStart(pod *v1.Pod,
 			if count == 1 {
 				continue
 			}
-			// if there is a reason to preserve the older container, do so
-			if _, ok := initContainersToKeep[status.ID]; ok {
+			// prune all other init containers that match this container name
+			glog.V(4).Infof("Removing init container %q instance %q %d", status.Name, status.ID.ID, count)
+			if err := m.DeleteContainer(status.ID); err != nil {
```

Comment:
You're right. Switching both to `m.removeContainer(status.ID.ID)` to make them consistent. I think we do need to clean up the log here since we don't do it anywhere else.